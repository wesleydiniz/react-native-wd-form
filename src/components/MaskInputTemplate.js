import React from "react";
import { View, Text } from "react-native";
import { TextInputMask } from "react-native-masked-text";

export default function maskedInputTemplate(locals) {
  if (locals.hidden) {
    return null;
  }

  const stylesheet = locals.stylesheet;
  let formGroupStyle = stylesheet.formGroup.normal;
  let controlLabelStyle = stylesheet.controlLabel.normal;
  let textboxStyle = stylesheet.textbox.normal;
  let helpBlockStyle = stylesheet.helpBlock.normal;
  const errorBlockStyle = stylesheet.errorBlock;

  if (locals.hasError) {
    formGroupStyle = stylesheet.formGroup.error;
    controlLabelStyle = stylesheet.controlLabel.error;
    textboxStyle = stylesheet.textbox.error;
    helpBlockStyle = stylesheet.helpBlock.error;
  }

  if (locals.editable === false) {
    textboxStyle = stylesheet.textbox.notEditable;
  }

  const label = locals.label ? (
    <Text style={controlLabelStyle}>{locals.label}</Text>
  ) : null;
  const help = locals.help ? (
    <Text style={helpBlockStyle}>{locals.help}</Text>
  ) : null;
  const error =
    locals.hasError && locals.error ? (
      <Text accessibilityLiveRegion="polite" style={errorBlockStyle}>
        {locals.error}
      </Text>
    ) : null;

  let maskType = "money";
  let maskOptions = {};
  if (locals.config) {
    if (locals.config.mask) {
      maskType = locals.config.mask;
    }

    if (locals.config.options) {
      maskOptions = locals.config.options;
    }
  }

  return (
    <View style={formGroupStyle}>
      {label}
      <TextInputMask
        type={maskType}
        options={maskOptions}
        accessibilityLabel={locals.label}
        ref="input"
        autoCapitalize={locals.autoCapitalize}
        autoCorrect={locals.autoCorrect}
        autoFocus={locals.autoFocus}
        blurOnSubmit={locals.blurOnSubmit}
        editable={locals.editable}
        keyboardType={locals.keyboardType}
        maxLength={locals.maxLength}
        multiline={locals.multiline}
        onBlur={locals.onBlur}
        onEndEditing={locals.onEndEditing}
        onFocus={locals.onFocus}
        onLayout={locals.onLayout}
        onSelectionChange={locals.onSelectionChange}
        onSubmitEditing={() => locals.onSubmitEditing()}
        placeholderTextColor={locals.placeholderTextColor}
        secureTextEntry={locals.secureTextEntry}
        selectTextOnFocus={locals.selectTextOnFocus}
        selectionColor={locals.selectionColor}
        numberOfLines={locals.numberOfLines}
        underlineColorAndroid={locals.underlineColorAndroid}
        clearButtonMode={locals.clearButtonMode}
        clearTextOnFocus={locals.clearTextOnFocus}
        enablesReturnKeyAutomatically={locals.enablesReturnKeyAutomatically}
        keyboardAppearance={locals.keyboardAppearance}
        onKeyPress={locals.onKeyPress}
        returnKeyType={locals.returnKeyType}
        selectionState={locals.selectionState}
        onChangeText={value => locals.onChange(value)}
        onChange={locals.onChangeNative}
        placeholder={locals.placeholder}
        style={textboxStyle}
        value={locals.value}
      />
      {help}
      {error}
    </View>
  );
}
