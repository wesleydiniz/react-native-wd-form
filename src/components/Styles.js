/**
 * @providesModule common-styles
 * @flow
 */
import { StyleSheet } from 'react-native';

export const COLORS = {
  BACKGROUND_COLOR: '#FFF',
  LIGHTER_COLOR: '#152C55',
  LIGHT_COLOR: '#4D648D',
  MEDIUM_COLOR: '#2D4571',
  DARK_COLOR: '#152C55',
  DARKEST_COLOR: '#051838',
  TEXT_COLOR: '#383838',
  TEXT_LIGHT_COLOR: '#dbdbdb'
};

export default StyleSheet.create({
  basicLabel: {
    color: COLORS.TEXT_COLOR,
    fontSize: 15,
    fontWeight: 'bold'
  },

  basicTitle: {
    color: COLORS.TEXT_COLOR,
    fontSize: 35,
    fontWeight: 'bold'
  },

  basicText: {
    color: COLORS.TEXT_COLOR,
    fontSize: 15
  },
  container: {
    flex: 1,
    backgroundColor: COLORS.BACKGROUND_COLOR
  },
  top: {
    flex: 1,
    padding: 10
  },
  content: {
    flex: 4,
    padding: 10
  },
  footer: {
    flex: 1,
    padding: 10
  }
});
