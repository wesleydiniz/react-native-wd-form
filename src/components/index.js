/**
 * @providesModule my-components
 */

import BasicButton from './BasicButton';
import BasicInput from './BasicInput';
import IconPicker from './IconPicker';
import IconInput from './IconInput';

export { BasicButton };
export { BasicInput };
export { IconPicker };
export { IconInput };
