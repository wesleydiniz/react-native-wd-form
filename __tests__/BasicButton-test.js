import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import { BasicButton } from '../src/components';

it('renders correctly', () => {
  const button = renderer.create(<BasicButton label="Test Button" />).toJSON();
  expect(button).toMatchSnapshot();
});
