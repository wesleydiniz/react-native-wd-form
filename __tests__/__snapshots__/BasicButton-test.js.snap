// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#152C55",
      "borderRadius": 2,
      "flexDirection": "row",
      "height": 50,
      "justifyContent": "center",
      "margin": 15,
      "opacity": 1,
      "paddingLeft": 10,
      "paddingRight": 10,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "alignItems": "center",
        "color": "#fff",
        "fontSize": 20,
        "fontWeight": "bold",
        "justifyContent": "center",
      }
    }
  >
    Test Button
  </Text>
</View>
`;
